### PR TITLE
feat: add allow missing field

### LINKED ISSUE #59: Syntax for specifying missing columns
Currently, a type specification of `Optional[int]` means that a column must be of integer type but may contain nulls.

We currently don't support a syntax to specify that it is allowed that a column is missing.

One current workaround is to specify `Foo.validate(df, allow_missing_columns=True)`, where `allow_missing_columns` is passed on to `_find_errors` as a kwarg (we should add this as an explicit parameter).

The following example contains a suggestion for how we _could_ allow missing columns (see `c`). It is one that @JakobGM came up with last year.
```python
import patito as pt
from typing import Optional

class Foo(pt.Model):
    a: int # only ints
    b: Optional[int] # mix ints and nulls
    c: int = None # column may be missing, but if it's there it must be an int - but this fails a type check
    d: Optional[int] = None # column may be missing, but if it's there it must be an int
```

An alternative would be to use `pt.Field` / `ColumnInfo`, and do something like the following, which I might like better, just because it will pass type checks.
```python
class Foo(pt.Model):
    c: int = pt.Field(allow_missing=True)
```

I am very open to ideas here. Does anyone have a suggestion? Tagging a few possibly-interested parties, @brendancooley, @dsgibbons, @ion-elgreco

### PR DESCRIPTION
**Pull request by dsgibbons**
Closes #59 

This PR adds a new field to `ColumnInfo` called `allow_missing`. During validation, if we find that a column is missing, we check if `allow_missing` is set inside `schema.column_infos[missing_column]`. If `True`, we don't append an error.

Note that we need to check whether `schema.column_infos.get(missing_column) is not None` because the user may have requested a superfluous column (e.g., `test_validators.test_validation_column_subset()`.

### CODE REVIEW

**Comment by munnmm** [comment]
This would be super helpful to have.

**Comment by dsgibbons** [comment]
Fix applied so `allow_missing` works with nested models.

**Comment by dsgibbons** [comment]
I want to expand the tests to include lists of structs - but I cannot add these tests until #100 has been merged.

**Comment by thomasaarholt** [comment]
A while back, @JakobGM and I spoke about using the following `= None` syntax to represent columns that are allowed to be missing:

```python
class Foo(pt.Model):
    id: str
    optional_column: Optional[str] = None
```

I think I prefer your suggestion, since it is way more explicit, and I really had to think to remember the above suggestion.

I'm happy to merge this, great if you can add a test to cover the (now merged) #100 optional list of structs.

**Comment by dsgibbons** [comment]
I believe the `allow_missing` syntax was your idea: https://github.com/JakobGM/patito/issues/59 - but yes, I prefer it too. 

Sounds good - I'll get those tests added.

**Comment by thomasaarholt** [comment]
Bloody hell, clearly the lack of sleep has gotten to me a bit 😂 thank you for implementing this!

**Comment by dsgibbons** [comment]
Tests updated and ready for review @thomasaarholt

**Review by thomasaarholt — APPROVED** [review]
Excellent work, and thank you again!

**Pull request merged**
